hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoGetAccountBalanceHandler.java

```diff
@@ -58,13 +77,84 @@ public Response createEmptyResponse(@NonNull final ResponseHeader header) {
     @Override
     public void validate(@NonNull final QueryContext context) throws PreCheckException {
         requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+        final var query = context.query();
+        final var accountStore = context.createStore(ReadableAccountStore.class);
+        final CryptoGetAccountBalanceQuery op = query.cryptogetAccountBalanceOrThrow();
+        if (op.hasAccountID()) {
+            final var account = accountStore.getAccountById(requireNonNull(op.accountID()));
+            if (account == null) throw new PreCheckException(INVALID_ACCOUNT_ID);
+            if (account.deleted()) {
+                throw new PreCheckException(ACCOUNT_DELETED);
```

Comment:
You can also use `validateTruePrecheck` method in `PrecheckException`

Reply:
Updated with validateFalsePreCheck.